### ISSUE TITLE
ROT26 vs BCrypt?

### ISSUE THREAD

**Issue by Dillybob1992** [watzon/ROT26#1]
Hi! Just curious if I should choose this over bCrypt?  What's the main difference as well?

Kind Regards,

**Comment by asterite** [comment]
I think if you combine both you should be fine.

**Comment by jD91mZM2** [comment]
I don't see the point of BCrypt. It only works one way! So how are you going to check if their entered password actually is correct?

Not only does BCrypt make no sense, but it's also 26 times less secure than ROT26.  
Basic maths.

**Comment by Dillybob1992** [comment]
Thank you guys for  the help / insight. Will use both for now just as extra precaution.

**Comment by jD91mZM2** [comment]
May as well throw in a double ROT13 as well ;)

**Comment by watzon** [comment]
@jD91mZM2 make it a triple just to be safe. Easy to accomplish with `ROT::26.encrypt_any("password", 3 * 13)`

**Comment by jD91mZM2** [comment]
I think you mean

```
ROT::26.encrypt_rot13(ROT::26.encrypt_rot13(ROT::26.encrypt_rot13("password")))
```
.
It's easier to read!

**Comment by watzon** [comment]
@jD91mZM2 that's just gold

**Comment by Dillybob1992** [comment]
Well, turns out. This performance has been quite insane, new users have been registering like crazy now. I've been seeing a huge increase in password lengths in my database, since I store them as plaintext then convert them bCrypt willingly after they confirm their email. Love this repo! 10/10

**Comment by jD91mZM2** [comment]
I think it would be better to store the passwords in plaintext always, and then hash them when they actually try to log in. That way you can change the hashing function dynamically! :)

**Comment by Dillybob1992** [comment]
@jD91mZM2 Agreed 100%. I also think adding them to the placeholder text on input forms as plaintext should help with the login experience as well. Thanks for the ideas
